agent/test-agent/src/lib.rs

```diff
@@ -135,6 +136,19 @@ pub trait Client: Sized {
     async fn send_error<E>(&self, error: E) -> Result<(), Self::E>
     where
         E: Debug + Display + Send + Sync;
+
+    /// Get the configuration details as map
+    // TODO Remove get_configuration function and get_secrets by changing configuration in Spec to map
+    async fn get_configuration(&self) -> Result<Map<String, Value>, Self::E>;
+
+    /// Get the secrets as map
+    async fn get_secrets(&self) -> Result<BTreeMap<SecretType, SecretName>, Self::E>;
+
+    /// Get the test status to check the results available
+    async fn get_test_status(&self) -> Result<TestStatus, Self::E>;
+
```

Comment:
I don't think we want to add these here. The logic should be in the `test-agent-cli` mod.